upcloud_api/cloud_manager/storage_mixin.py

```diff
@@ -184,33 +184,43 @@ def create_storage_import(
         Creates an import task to import data into an existing storage.
         Source types: http_import or direct_upload.
         """
+        if source not in ["http_import", "direct_upload"]:
+            raise Exception("invalid storage import source: %s", source)
```

Comment:
You're initializing an exception with two args – you probably want to `f`ormat only one.

```suggestion
        if source not in ("http_import", "direct_upload"):
            raise Exception(f"invalid storage import source: {source}")
```